src/FakeItEasy/Core/FakeScope.cs

```diff
@@ -110,11 +110,11 @@ public override IEnumerator<ICompletedFakeObjectCall> GetEnumerator()
 
             internal override IEnumerable<ICompletedFakeObjectCall> GetCallsWithinScope(FakeManager fakeObject)
             {
-                List<ICompletedFakeObjectCall> calls;
+                IList<ICompletedFakeObjectCall> calls;
 
                 if (!this.recordedCallsGroupedByFakeManager.TryGetValue(fakeObject, out calls))
                 {
-                    calls = new List<ICompletedFakeObjectCall>();
+                    calls = new SynchronizedCollection<ICompletedFakeObjectCall>();

```

Comment:
Actually, we should probably return `Enumerable.Empty<T>()` here... the method returns `IEnumerable<T>`, so the result is not meant to be modified by the caller.